### PR TITLE
feat: replace lookmlcodeblock with codedisplay

### PR DESCRIPTION
**Pull request by noahamac**
Adds in codedisplay and addresses sqlTableName as promised.

### CODE REVIEW

File: src/utils/LookmlDiagrammer/diagrammer.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export function generateExploreDiagram(explore: ILookmlModelExplore, hiddenToggl
     }).length
 
     const firstSetName = explore.sets[0].name
-    if (explore.name === viewName || firstSetName === viewName || firstSetName.split(".")[0] === viewName) {
+    if (baseViewName === "" && (explore.name === viewName || firstSetName === viewName || firstSetName.split(".")[0] === viewName)) {
```
Comment (noahamac) [inline src/utils/LookmlDiagrammer/diagrammer.ts:109]: Unrelated fix I'm sneaking in here

File: src/components/DiagramFrame/MetadataPanel/utils.ts [hunk 2]
```diff
@@ -117,15 +119,17 @@ export function getFieldMetadata(fields: ILookmlModelExploreField[], selectionIn
 }
 
 export function getViewMetadata(viewResponse: ILookmlModelExplore, isLoading: boolean, lookmlLink: string, selectionInfo: SelectionInfoPacket) {
-  let sqlTableName = "unknown"
-  if (isLoading) {
-    sqlTableName = "Loading..."
-  } else if (viewResponse?.name === selectionInfo.name) {
+  let sqlTableName: string
```
Comment (noahamac) [inline src/components/DiagramFrame/MetadataPanel/utils.ts:122]: I suppose this is still overloading... But it's better?

Comment (noahamac) [reply src/components/DiagramFrame/MetadataPanel/utils.ts:122]: I should use the APIX pattern here where I pass back renderers... But that's a refactor for another day :D

**Comment by noahamac** [comment]
Gotta merge #27 first

File: src/components/DiagramFrame/MetadataPanel/LookmlCodeBlock.tsx [hunk 3]
```diff
@@ -24,14 +24,81 @@
...
+                 {line.map((token, key) => {
+                   const tokenProps = getTokenProps({ token, key })
+                   const text = tokenProps.children
+                   if (lookmlKeywords.includes(text.trim())) {
+                     tokenProps.style = {
+                       ...tokenProps.style,
+                       color: "rgb(12, 150, 155)"
+                     }
+                   }
+                   return <span key={key} {...tokenProps} />
```
Comment (bryans99) [inline src/components/DiagramFrame/MetadataPanel/LookmlCodeBlock.tsx:94]: Not obvious code here. You have a span with no obvious children but there is. Consider something like (I have not tested though)
```
const { children: text, ...tokenProps } = getTokenProps({ token, key })
if (lookmlKeywords.includes(text.trim())) {
   tokenProps.style = {
     ...tokenProps.style,
     color: "rgb(12, 150, 155)"
   }
 }
return <span key={key} {...tokenProps} >{text}</span>
```

File: src/components/DiagramFrame/MetadataPanel/utils.ts (hunk 2 above)
Comment (bryans99) [reply src/components/DiagramFrame/MetadataPanel/utils.ts:122]: Can live with it for now

File: src/utils/LookmlDiagrammer/diagrammer.ts (hunk 1 above)
Comment (bryans99) [reply src/utils/LookmlDiagrammer/diagrammer.ts:109]: || firstSetName.startsWith(viewName + ".")

**Review by bryans99 — COMMENTED** [review]
Consider changing a couple of things. If you disagree let me know and  I will approve as is

File: src/components/DiagramFrame/MetadataPanel/LookmlCodeBlock.tsx [hunk 4]
```diff
@@ -24,14 +24,81 @@
...
+                 {line.map((token, key) => {
+                   const tokenProps = getTokenProps({ token, key })
+                   const text = tokenProps.children
+                   if (lookmlKeywords.includes(text.trim())) {
+                     tokenProps.style = {
+                       ...tokenProps.style,
+                       color: "rgb(12, 150, 155)"
+                     }
+                   }
+                   return <span key={key} {...tokenProps}>{text}</span>
```
Comment (noahamac) [inline src/components/DiagramFrame/MetadataPanel/LookmlCodeBlock.tsx:94]: Good call

File: src/utils/LookmlDiagrammer/diagrammer.ts [hunk 5]
```diff
@@ -106,7 +106,7 @@ export function generateExploreDiagram(explore: ILookmlModelExplore, hiddenToggl
     }).length
 
     const firstSetName = explore.sets[0].name
-    if (explore.name === viewName || firstSetName === viewName || firstSetName.split(".")[0] === viewName) {
+    if (baseViewName === "" && (explore.name === viewName || firstSetName === viewName || firstSetName.startsWith(viewName + "."))) {
```
Comment (noahamac) [inline src/utils/LookmlDiagrammer/diagrammer.ts:109]: More readable

File: src/components/DiagramFrame/MetadataPanel/LookmlCodeBlock.tsx (hunk 4 above)
Comment (bryans99) [reply src/components/DiagramFrame/MetadataPanel/LookmlCodeBlock.tsx:94]: At this point tokenProps still contains the children property. The solution I outlined:
`const { children: text, ...tokenProps } = getTokenProps({ token, key })`
Should remove the children property from tokenProps. At the end of the day it probably doesnt hurt but best to be clear. You should also check my code. I think it works but you never know.

**Review by bryans99 — COMMENTED** [review]
See comment. I can approve if you disagree

**Comment by noahamac** [comment]
Yeah, you're right. And it does work still. Sorry about that, had a Thursday moment :)

**Review by bryans99 — APPROVED** [review]
LGTM
